hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java

```
@@ -359,6 +360,20 @@ public Builder<NodeStakeUpdateTransactionBody.Builder> nodeStakeUpdate() {
         return new Builder<>(TransactionType.NODESTAKEUPDATE, builder);
     }
 
+    public Builder<RandomGenerateTransactionBody.Builder> randomGenerate(int range) {
+        var builder = RandomGenerateTransactionBody.newBuilder().setRange(range);
+        var transactionBodyBuilder = new Builder<>(TransactionType.RANDOMGENERATE, builder);
+        return range < 0 ?
```

if `range < 0`, the transaction should fail with ` INVALID_RANDOM_GENERATE_RANGE = 324` and you need to set  `TransactionReceipt.status`.

---

I added some new tests, one for a negative range and one for `INVALID_RANDOM_GENERATE_RANGE`. I made the setting of the receipt status in the handler test; but I see that it could also be set in the `RecordItemBuilder`, and there may be precedents for setting the status in either place. Let me know if it looks good as is, or if I should fine tune it some more.

---

I mean in in `RecordItemBuilder`, you should do:

```java
    return range < 0 ? transactionBodyBuilder.status(ResponseCodeEnum. INVALID_RANDOM_GENERATE_RANGE) : ...
```

---

Up until now, we haven't put error scenarios in `RecordItemBuilder`. Can't we just do this in the one test that needs to test this?